src/main/java/de/dennisguse/opentracks/content/data/Layout.java

```diff
@@ -1,16 +1,61 @@
 package de.dennisguse.opentracks.content.data;
 
+import android.content.res.Resources;
+import android.os.Parcel;
+import android.os.Parcelable;
+
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
+import java.util.stream.Collectors;
+
+import de.dennisguse.opentracks.R;
+import de.dennisguse.opentracks.settings.PreferencesUtils;
+import de.dennisguse.opentracks.util.CsvConstants;
 
-public class Layout {
+public class Layout implements Parcelable {
     private final String profile;
+    private int columnsPerRow;
     private final List<DataField> dataFields = new ArrayList<>();
 
+    public static Layout fromCsv(String csvLine, Resources resources) {
+        List<String> csvParts = Arrays.asList(csvLine.split(CsvConstants.ITEM_SEPARATOR));
+        Layout layout = new Layout(csvParts.get(0), Integer.parseInt(csvParts.get(1)));
+        for (int i = 2; i < csvParts.size(); i++) {
+            String[] fieldParts = csvParts.get(i).split(CsvConstants.PROPERTY_SEPARATOR);
```

Comment:
Index out of bounds check.

Reply:
Same here: https://github.com/OpenTracksApp/OpenTracks/pull/1012#discussion_r763366589